From 078459f409ab9fec4f8bf89b8daf2c776932bf13 Mon Sep 17 00:00:00 2001
From: Jiajin Cui <cuijiajin@uniontech.com>
Date: Fri, 10 Jul 2026 19:03:09 +0800
Subject: [PATCH 179/268] winewayland: Fix y coordinate of IME cursor
 rectangle.

The second argument of zwp_text_input_v3_set_cursor_rectangle() is the y
coordinate, but surface_rect.right was passed instead of surface_rect.top,
placing the IME candidate window at the wrong vertical position.

Signed-off-by: Jiajin Cui <cuijiajin@uniontech.com>
---
 dlls/winewayland.drv/wayland_text_input.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winewayland.drv/wayland_text_input.c b/dlls/winewayland.drv/wayland_text_input.c
index 3d865ef5492..5b78f8db8d9 100644
--- a/dlls/winewayland.drv/wayland_text_input.c
+++ b/dlls/winewayland.drv/wayland_text_input.c
@@ -250,7 +250,7 @@ BOOL WAYLAND_SetIMECompositionRect(HWND hwnd, RECT rect)
     wayland_win_data_release(data);
 
     zwp_text_input_v3_set_cursor_rectangle(text_input->zwp_text_input_v3,
-            surface_rect.left, surface_rect.right, surface_rect.right - surface_rect.left,
+            surface_rect.left, surface_rect.top, surface_rect.right - surface_rect.left,
             surface_rect.bottom - surface_rect.top);
     zwp_text_input_v3_commit(text_input->zwp_text_input_v3);
 
-- 
2.55.0

